### PR TITLE
os/bluestore: slightly refactor Blob::try_reuse_blob

### PR DESCRIPTION
**Pull request by xiexingguo**
(1) rename try_reuse_blob->can_reuse_blob to keep pace with
can_split/can_split_at.
(2) replace blob with get_blob().

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

**Comment by xiexingguo** [comment]
@ifed01 Igor, I slightly refactor the try_reuse_blob method.

Mind having a look?

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -1712,7 +1712,7 @@ void BlueStore::Blob::get_ref(
   // references.  Otherwise one is neither unable to determine required
   // amount of counters in case of per-au tracking nor obtain min_release_size
   // for single counter mode.
-  assert(get_blob().get_logical_length() != 0);
+  assert(blob.get_logical_length() != 0);
```
Comment (ifed01) [inline src/os/bluestore/BlueStore.cc]: Actually I'd prefer the reverse change - replace all direct blob access with corresponding get_blob/dirty_blob calls. Originally I intended to distinguish read-only/read-write access to a blob by using them. It seems useful here as direct 'write' access to a blob might be erroneous - as it lacks blob_bl cleanup.  Unfortunately we don't have corresponding means in C++ for access class properties from within class members (well - in fact we can define a parent class and put blob and access methods there but that's probably an overkill). Hence this evolved to some mixture that worth a cleanup.

Comment (ifed01) [inline src/os/bluestore/BlueStore.cc]: Others look good.

**Comment by xiexingguo** [comment]
@ifed01  Updated! Can you recheck?

**Comment by xiexingguo** [comment]
retest this please

**Comment by xiexingguo** [comment]
retest this please

**Comment by ifed01** [comment]
and please replace "(2) use blob directly." in the commit message...

**Pull request merged**